ludwig/models/modules/convolutional_modules.py

```diff
@@ -530,74 +535,57 @@ def __init__(
-                ]
-                layer_output = conv_2d_layer(hidden, filter_shape,
-                                             [layer['num_filters']],
-                                             stride=layer['stride'],
-                                             padding='SAME',
-                                             activation=layer['activation'],
-                                             norm=layer['norm'],
-                                             dropout=layer['dropout'],
-                                             dropout_rate=dropout_rate,
-                                             regularizer=regularizer if layer[
-                                                 'regularize'] else None,
-                                             initializer=layer['initializer'],
-                                             is_training=is_training)
-                prev_num_filters = layer['num_filters']
-                logger.debug('  conv_{}: {}'.format(i, layer_output))
+        self.stack = []
 
-                # Pooling
+        for i, layer in enumerate(self.layers):
+            with tf.variable_scope('conv_' + str(i)):
+                # TODO add filter size tuples
+                # TODO provide regularization penalty
+                # TODO provide regularizers
+                # TODO add initializers
+                self.stack.append(
+                    Conv2D(
+                        layer['num_filters'],
+                        layer['filter_size'],
+                        # kernel_regularizer=regularizer if layer['regularize'] else None,
+                    )
+                )
+
+                self.stack.append(Activation(layer['activation']))
+
+                if layer['dropout_rate'] > 0:
+                    self.stack.append(Dropout(layer['dropout_rate']))
+                
                 if layer['pool_size'] is not None:
-                    pool_size = layer['pool_size']
-                    pool_kernel_size = [1, pool_size, pool_size, 1]
-                    pool_stride = layer['pool_stride']
-                    if pool_stride is not None:
-                        pool_kernel_stride = [1, pool_size, pool_size, 1]
-                    else:
-                        pool_kernel_stride = [1, pool_stride, pool_stride, 1]
-                    layer_output = tf.nn.max_pool(
-                        layer_output,
-                        ksize=pool_kernel_size,
-                        strides=pool_kernel_stride,
-                        padding='SAME',
-                        name='pool_{}'.format(i)
+                    self.stack.append(
+                        MaxPool2D(
+                            pool_size=layer['pool_size'],
+                            strides=layer['pool_strides']
+                        )
                     )
 
-                    logger.debug('  pool_{}: {}'.format(
-                        i, layer_output))
+    def __call__(
```

Comment:
ConvStack2D should extend Layer and because of that this should be `call()` instead of `__call__()`.
Also regularizer and dropout rate shoudl be removed and a `mask=None` parameter shoudl be added (because `Layer.call()` expects it and because it should be propagated to the internal layers calls too.

Reply:
Updated.